bidscoin/plugins/pet2bids.py

```diff
@@ -75,32 +75,36 @@ def is_sourcefile(file: Path) -> str:
     #         return 'PETXLS'
 
     if bids.is_dicomfile(file):
+        # @Anthony: I don't think the first if-statement is the right way to go as BIDScoin always determines if a file is a sourcefile first, and only then tries to read attributes from it
         if 'pt' in str.lower(get_attribute('DICOM', file, 'Modality')):
-            return 'PET'
-        elif 'pt' == str.lower(pydicom.dcmread(file).Modality):
-            return 'PET'
+            return 'DICOM'
+        elif 'pt' == bids.get_dicomfield('Modality', file).lower():
+            return 'DICOM'
 
     return ''
 
 
 def get_attribute(dataformat: str, sourcefile: Path, attribute: str, options: dict = {}) -> Union[str, int]:
     """
-    This plugin supports reading attributes from the PET Excel file
+    This plugin supports reading attributes from the PET Excel sidecar file
 
-    :param dataformat:  The bidsmap-dataformat of the sourcefile, e.g. PETXLS
+    :param dataformat:  The bidsmap-dataformat of the sourcefile, e.g. DICOM
     :param sourcefile:  The sourcefile from which the attribute value should be read
     :param attribute:   The attribute key for which the value should be read
     :param options:     A dictionary with the plugin options, e.g. taken from the bidsmap['Options']
     :return:            The attribute value
     """
-    # if dataformat == 'PETXLS':
-    #
-    #     data = pet.helper_functions.single_spreadsheet_reader(sourcefile)
-    #
-    #     return data.get(attribute)
 
-    if dataformat == 'PET' and bids.is_dicomfile(sourcefile):
+    for ext in options['meta']:
+        if sourcefile.with_suffix(ext).is_file():
```

Comment:
I don't know if this would work for you like this? Reading xls metadata can be done from here, but then it is specific for the pet2bids plugin. From BIDScoin's perspective having a source sidecar file with the same name is what I call "extended attributes" and, if you implement your xls reader as such (-> bids.DataSource._extattributes()), then it would be available throughout BIDScoin

Reply:
To be honest, I'm not 100% sure on how to implement the spreadsheet portion for PET. But I am sure that I'm going to need an `*.is_spreadsheetfile()` or `*.is_pet_spreadsheetfile()` method to determine if it's PET or not based on what's in the header row.

Then `if pet spreadsheet is True` -> `associate pet spreadsheet w/ imaging data`. It's the later part that's fuzzy on me with respect to BIDSCoin. But, I don't think it's entirely necessary at this point as one can add everything required to make a BIDS Valid PET with the Mapper and Editor as of right now.

Reply:
Mhh, I have no idea what your xls content looks like, can you give me some idea (screenshots or so)? Do you have a different xls for every subject/measurement?

I am happy to give you a hand with getting the xls data into BIDScoin in the best way